### PR TITLE
Enable running revert migrations when downgrading using ipfs-update

### PR DESCRIPTION
**Pull request by hsanjuan**

### CODE REVIEW

**Comment by hsanjuan** [comment]
Is this something we were not doing for a reason?

**Comment by Stebalien** [comment]
We might want to put this behind a flag.

**Comment by hsanjuan** [comment]
Probably the whole downgrade should be behind a flag, not just the migrations?

**Comment by Stebalien** [comment]
A migration is a bigger deal as it can sometimes take a while. But yeah, just putting everything behind a flag is probably a reasonable approach.

File: main.go [hunk 1]
```diff
@@ -120,6 +120,10 @@ var cmdInstall = cli.Command{
 			Name:  "no-check",
 			Usage: "Skip pre-install tests.",
 		},
+		cli.BoolFlag{
+			Name:  "downgrade",
```
Comment (Stebalien) [inline main.go:124]: `--downgrade-ok` or `--allow-downgrade`. When I see `--downgrade`, I think that downgrades are the _only_ thing we allow.

File: main.go [hunk 2]
```diff
@@ -120,6 +120,10 @@ var cmdInstall = cli.Command{
...
+			Usage: "Allow downgrading (and reverting any migrations).",
```
Comment (Stebalien) [inline main.go:125]:
```suggestion
			Usage: "Allow downgrading. WARNING: Downgrades may require running reverse migrations.",
```

File: main.go [hunk 3]
```diff
@@ -193,10 +202,16 @@ var cmdRevert = cli.Command{
 	Name:      "revert",
 	Usage:     "Revert to previously installed version of ipfs.",
 	ArgsUsage: " ",
-	Description: `'revert' will check if a previous update left a stashed
+	Description: `
```
Comment (Stebalien) [inline main.go:205]: This will add an extra newline, right?

File: main.go [hunk 4]
```diff
@@ -193,10 +202,16 @@ var cmdRevert = cli.Command{
...
+   'revert' will check if a previous update left a stashed
    binary and overwrite the current ipfs binary with it.
+
+   Using 'revert' will not run any datastore migrations. For that, use
+   'ipfs-update install --downgrade <prev-version>'.
```
Comment (Stebalien) [inline main.go:210]: Hm. I'd _expect_ revert to _always_ run migrations. I'd expect it to be a "undo what I just did" command.

File: main.go (hunk 3 above)
Comment (hsanjuan) [reply main.go:205]: It does not look bad but ok.

File: main.go (hunk 4 above)
Comment (hsanjuan) [reply main.go:210]: Yeah, I did not do that, but I'd expect the same and thought about warning about it.

On the other hand, it is good to have a way to downgrade without running migrations, but this is not the best. We should fix this on a separate PR though.

Comment (hsanjuan) [reply main.go]: https://github.com/ipfs/ipfs-update/issues/127

**Comment by Stebalien** [comment]
@hsanjuan could you also cut a release?

**Comment by hsanjuan** [comment]
yes yes

**Pull request merged**